File: t6/Triangle.cpp
/**
 * Class that defines a triangle.
 *
 * @author João Bolsson (jvmarques@inf.ufsm.br)
 * @since 2017, 17 May.
 */

#include "Shape.cpp"

class Triangle : public Shape {

private:
    int base, height;
public:

    /**
     * Default construct for a simple triangle.
     *
     * @param base Triangle base.
     * @param height Triangle height.
     */
    Triangle(int base, int height) {
        this->base = base;
        this->height = height;
    }

    /**
     * @return Triangle area.
     */
    double getArea() {
        return (base * height) / 2;
    }
};

File: t6-java/src/main/Triangle.java
package main;

/**
 * Class that defines a triangle.
 *
 * @author João Bolsson (joaovictorbolsson@gmail.com)
 * @since 2017, 27 May.
 */
public class Triangle extends Shape {

    private final int base, height;

    /**
     * Default construct.
     *
     * @param base Triangle base.
     * @param height Triangle height.
     */
    public Triangle(final int base, final int height) {
        this.base = base;
        this.height = height;
    }

    /**
     * @return Triangle area.
     */
    @Override
    public double getArea() {
        return (base * height) / 2;
    }

    @Override
    public String toString() {
        return "Triângulo";
    }

}
